src/parlant/core/engines/alpha/engine.py

```diff
@@ -396,6 +438,106 @@ async def _do_process(
                 },
             )
 
+    async def _do_utter(
+        self,
+        context: Context,
+        interaction: _InteractionState,
+        event_emitter: EventEmitter,
+        actions: Sequence[UtteranceRequest],
+    ) -> None:
+        agent = await self._agent_store.read_agent(context.agent_id)
+        session = await self._session_store.read_session(context.session_id)
+        customer = await self._customer_store.read_customer(session.customer_id)
+
+        await event_emitter.emit_status_event(
+            correlation_id=self._correlator.correlation_id,
+            data={
+                "acknowledged_offset": interaction.last_known_event_offset,
+                "status": "acknowledged",
+                "data": {},
+            },
+        )
+
+        try:
+            context_variables = await self._load_context_variables(
+                agent_id=context.agent_id, session=session, customer=customer
+            )
+
+            terms = set(
+                await self._load_relevant_terms(
+                    agents=[agent],
+                    context_variables=context_variables,
+                )
+            )
+
+            ordinary_guideline_propositions = await self._utter_requests_to_guideline_propositions(
+                actions
+            )
+
+            await event_emitter.emit_status_event(
+                correlation_id=self._correlator.correlation_id,
+                data={
+                    "acknowledged_offset": interaction.last_known_event_offset,
+                    "status": "processing",
```

Comment:
Status type here should be "typing" not "processing", since we are literally just typing a message and not running any tools.